cmd/cli/devstack/devstack.go

```diff
@@ -144,10 +143,20 @@ func runDevstack(cmd *cobra.Command, ODs *devstack.DevStackOptions, IsNoop bool)
 
 	cm := util.GetCleanupManager(ctx)
 
-	fsRepo, err := repo.NewFS(viper.GetString("repo"))
+	repoPath, _ := os.MkdirTemp("", "")
+	cm.RegisterCallback(func() error {
+		// We need to clean up the repo when the node shuts down, but we can ONLY
+		// do this because we know it is a temporary directory.
+		return os.RemoveAll(repoPath)
+	})
```

Comment:
Since this command blocks until its done we could just defer this, trying to move away from the cleanup manager

Reply:
Ooh good call